### PR TITLE
Xch/snprintf vs snprintk

### PR DESCRIPTION
**Pull request by XavierChapron**
This is a first draft of changes that could avoid linking both snprintk andsnprintf when possible.
It is linked to https://github.com/zephyrproject-rtos/zephyr/issues/24197.

I have for now focus only on `drivers` and `subsys` folders.

They are a few snprintf that can't be easily changed:

- `shell_fprintf.c`: `shell_fprintf_fmt()` but I'm not sure about the #ifdef to choose between `z_prf` and `z_vprintk`.
- `lwm2m_rw_plain_text.c`: `plain_text_put_float64fix()` as printk doesn't support printing %lld values.
- All the ones that I've missed.

This is obviously just a start, that might be enough or not depending on the modules a project uses.

### CODE REVIEW

File: drivers/ethernet/eth_native_posix_adapt.c [hunk 1]
```diff
@@ -95,7 +95,7 @@ static int ssystem(const char *fmt, ...)
 	int ret;
 
 	va_start(ap, fmt);
-	vsnprintf(cmd, sizeof(cmd), fmt, ap);
+	vsnprintk(cmd, sizeof(cmd), fmt, ap);
```
Comment (jukkar) [inline drivers/ethernet/eth_native_posix_adapt.c:98]: Nack to this specific change. The `vsnprintf()` is actually a call to Linux libc here.

**Review by jukkar — CHANGES_REQUESTED** [review]
+1 to changing the code to use `snprintk()`, there is just one change in native_posix eth driver that needs to be changed. Other than that, the networking related changes look ok.

File: drivers/ethernet/eth_native_posix_adapt.c (hunk 1 above)
Comment (XavierChapron) [reply drivers/ethernet/eth_native_posix_adapt.c]: Removed!

**Review by jukkar — APPROVED** [review]
+1 to networking changes

**Review by tbursztyka — APPROVED** [review]

File: subsys/fs/fuse_fs_access.c [hunk 2]
```diff
@@ -496,17 +496,19 @@ static void fuse_fs_access_init(void)
...
 	if (fuse_mountpoint == NULL) {
 		return;
 	}
 
-	full_cmd = malloc(strlen(cmd) + strlen(fuse_mountpoint) + 1);
+	full_cmd_length = strlen(cmd) + strlen(fuse_mountpoint) + 1;
+	full_cmd = malloc(full_cmd_length);
 
-	sprintf(full_cmd, "%s%s", cmd, fuse_mountpoint);
+	snprintk(full_cmd, full_cmd_length, "%s%s", cmd, fuse_mountpoint);
```
Comment (vanwinkeljan) [inline subsys/fs/fuse_fs_access.c:509]: Note that this function is called outside of the zephyr kernel context so no zephyr specific calls  should be made

File: subsys/fs/fuse_fs_access.c [hunk 3]
```diff
@@ -66,7 +66,7 @@ static bool is_mount_point(const char *path)
 {
 	char dir_path[PATH_MAX];
 
-	sprintf(dir_path, "%s", path);
+	snprintk(dir_path, sizeof(dir_path), "%s", path);
```
Comment (vanwinkeljan) [inline subsys/fs/fuse_fs_access.c:69]: note that this function is called from the host machine and not in zephyr context

**Review by vanwinkeljan — DISMISSED** [review]
fuse related changes should not be applyed as these calls happen from host

File: subsys/fs/fuse_fs_access.c (hunk 2 above)
Comment (XavierChapron) [reply subsys/fs/fuse_fs_access.c]: Removed

File: subsys/fs/fuse_fs_access.c (hunk 3 above)
Comment (XavierChapron) [reply subsys/fs/fuse_fs_access.c]: Removed

**Comment by carlescufi** [comment]
@XavierChapron please rebase this and fix the Gitlint issue if you want this in for 2.3

**Review by rlubos — APPROVED** [review]
OpenThread/LWM2M LGTM

**Review by tyrantlizard77 — APPROVED** [review]
thumbs up on having drivers and core kernel subsystems all use snprintk()

**Review by pfalcon — APPROVED** [review]

**Review by loicpoulain — APPROVED** [review]

**Comment by carlescufi** [comment]
I will merge this although the commit message is 
Issing a description, since the commits are very self-explanatory. @XavierChapron please add a description next time.

**Comment by XavierChapron** [comment]
@carlescufi Sorry I wasn't connected during the week-end.

**Pull request merged**